db/c.cc

```diff
@@ -341,6 +351,196 @@ struct rocksdb_compactionfilter_t : public CompactionFilter {
   bool IgnoreSnapshots() const override { return ignore_snapshots_; }
 };
 
+/* Table Properties */
+
+struct rocksdb_table_properties_t {
+  const TableProperties* rep;
+};
+
+struct rocksdb_table_properties_collector_factory_context_t {
+  TablePropertiesCollectorFactory::Context* rep;
+};
+
+struct rocksdb_user_collected_properties_t {
+  const UserCollectedProperties* rep;
+};
+
+struct rocksdb_user_collected_properties_mut_t {
+  UserCollectedProperties* rep;
+};
```

Comment:
This may be overkill; I introduced the wrapper above and had to cast away the `const`-ness to make it writable for `rocksdb_user_collected_properties_insert`. The alternatives are either accept some cheekyi casting, or sling bare void pointers to the C++ struct around, neither of which seems very appealing.